libraries/chain/deep_mind.cpp

```diff
@@ -202,6 +202,12 @@ namespace eosio::chain {
    {
       ++_action_id;
    }
+   void deep_mind_handler::on_call()
+   {
+      fc_dlog(_logger, "CREATION_OP CALL ${action_id}",
+         ("action_id", _action_id)
+      );
+   }
```

Comment:
This indicates which sync calls were called in the context of the current action. It does not provide any information on the nesting of the sync calls. Nor does it allow the deep-mind consumer to relate table actions to individual sync calls. 

Seems you need a mapping of sync calls into the transaction trace data. Currently `_action_id` provides that mapping for current `${action_id}` to a corresponding action trace of the transaction trace.

This might be as simple as having an `on_end_call() { ++_action_id; }` OR maybe it is having a completely separate `_call_id` and then modifying things like `on_create_table` to include a `${call_id}`

Reply:
This does raise the question if table changes are desired at the level of sync calls. Seems like they would be otherwise maybe we don't add sync calls to deep-mind at all.

Reply:
Adding sync_call level info for all table modifications will potentially require work from all users of substreams. Or substreams will need to be able to allow the user to determine they want all sync call table modifications rolled-up to the action level.

Reply:
I initially thought about using call_id to show sync call hierarchy within an action and stated that in the issue description. But then when I looked into inline action and require_recipient, I saw they only log the `action_id`. That's why I changed to log `action_id` only.